### DISCUSSION TITLE
Class Distribution Problem with OverSampling

### DISCUSSION THREAD

**Discussion by kilickursat** [pycaret/pycaret#2278] [Q&A]
I have tried to apply random oversampling using PyCaret and sklearn.  However, PyCaret provided imbalanced distribution instead of the majority. What is the reason for it? 

**PyCaret**
`from imblearn.over_sampling import RandomOverSampler ros = RandomOverSampler(random_state=42)`

`Session_2 = setup(df2, target = 'Layers', session_id=177, log_experiment=False, 
                  experiment_name='lithologies2', normalize=True, normalize_method='minmax', 
                  transformation=True, transformation_method = 'quantile', fix_imbalance=True,
                  fix_imbalance_method= ros)`

`df2.Layers.value_counts()`

`#Output; VSG    498,
VG     116, VCS     39`

**Sklearn**

`from imblearn.over_sampling import RandomOverSampler ros = RandomOverSampler(sampling_strategy="not majority")
X_ros, y_ros = ros.fit_resample(Xtrain, y_train)`

`#OUTPUT
2    392
1    392
0    392`
